articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md

```diff
@@ -0,0 +1,159 @@
+documentationcenter: dev-center-name
+author: andretms
+manager: mbaldwin
+editor: ''
+
+ms.service: active-directory
+ms.devlang: na
+ms.topic: article
+ms.tgt_pltfrm: na
+ms.workload: identity�
+ms.date: 05/09/2018
+ms.author: andret
+
+---
+
+# Sign-in users and acquire an access token from a JavaScript application
+
+This quickstart contains a code sample that demonstrates how a JavaScript Single Page Application (SPA) can sign in personal, work and school accounts, get an access token, and call the Microsoft Graph API.
+
+![How the sample app generated by this Quickstart works](media/active-directory-javascriptspa/javascriptspa-intro.png)
+
+> [!div renderon="docs"]
+> ## Register your application and download your Quickstart app
+> You have two options to start your Quickstart application:
+> ### Option 1: Register your application and download your Quickstart app [Express]
+>
+> 1. Go to the [Azure Portal - Application Registration [Prod]](https://portal.azure.com/signin/index/?Microsoft_AAD_RegisteredApps=true#blade/Microsoft_AAD_RegisteredApps/applicationsListBlade/quickStartType/JavascriptSpaQuickstartPage/sourceType/docs)
+> 1. Enter a name for your application and click **Register**
+> 1. Follow the instructions to download and configure your new application, adding **http<span/>://localhost:30662/** as Redirect URL and setting the **Implict Flow** automatically for you.
+> [!div renderon="docs"]
+> ### Option 2: Register your application and configure sample [Manual]
+> #### Step 1: Register your application
+> 
+> 1. To register an application, go to the [Azure Portal - Application Registration [Prod]](https://aka.ms/registeredappsprod) and select **New registration**
+> 1. Enter a name for your application, add **http<span/>://localhost:30662/** in Reply URL, and click **Register**
+> 1. Select **Authentication** menu and set **ID tokens** under *Implict Grant* and select 'Save'
+
+> [!div class="sxs-lookup" renderon="portal"]
+> ### Step 1: Configure your application in Azure Portal
+> For the code sample for this quickstart to work, you need to add a reply URL as **http<span/>://localhost:30662/** and enable **Implict Grant**.
+> > [!div renderon="portal" id="makechanges" class="nextstepaction"]
+> > [Make these changes for me]()
+>
+> > [!div id="appconfigured" class="hidden"]
+> > ![Already configured](media/active-directory-windesktop/checkmark.png) Your application is configured with these attributes
+
+> [!NOTE]
+> If you use Visual Studio, the Redirect URL will be set to to *http:<span/>//localhost:30662/* as it is configured in the code sample's project. If you use Python or any other web server, set redirect URL to *http://<span/>localhost:30662/*, run the following command line:
+>> ```bash
+>> python -m http.server 30662
+>> ```
+
+### Step 2: Download your web server or project
+
+- [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
```

Comment:
I think we shouldn't have any VS dependency. We have created dev samples in MSAL that doesn't require any Visual Studio dependency and just requires node.js.

Reply:
I agree with this...when the first thing I see is "download the VS project" I get turned off right away feeling...oh its complicated ;-).  I think we should have it but we have a nodejs version as well.

Reply:
@asnow003 - check my comment below